Emma/emma_libs/memoryManager.py

```diff
@@ -208,87 +198,116 @@ def createStandardReports():
-                    if element[Element.AddressStart] < startPoint:
+                    if len(str(element[Element.addressEnd])) > 11 or len(str(originalStartAddress)) > 11:          # Check if address start / end fits in the rectangle, 11 (pixels) is the size of the rectangle
+                        yAxe = yAxe + len(str(element[Element.addressEnd])) - 10
+                    if element[Element.addressStart] < startPoint:
                         xAxeRectStart = 0
-                        rectLength = element[Element.AddressEnd] - startPoint
-                        originalStartAddress = element[Element.OriginalAddressStart]
+                        rectLength = element[Element.addressEnd] - startPoint
+                        originalStartAddress = element[Element.originalAddressStart]
                         fontColour = "red"
                     if index == 0:
-                        biggestEndAddress = element[Element.AddressEnd]
+                        biggestEndAddress = element[Element.addressEnd]
                     else:
-                        # check if the actual element is overlapped by the last element
-                        if element[Element.AddressStart] <= elementstoPlot[index-1][Element.AddressStart] or element[Element.AddressStart] < elementstoPlot[index-1][Element.AddressEnd]:
+                        # Check if the actual element is overlapped by the last element
+                        if element[Element.addressStart] <= elementsToPlot[index - 1][Element.addressStart] or element[Element.addressStart] < elementsToPlot[index - 1][Element.addressEnd]:
                             yAxe = yAxe + distanceBetweenElements
-                            plottedElements.append((element[Element.AddressEnd], yAxe))
+                            plottedElements.append((element[Element.addressEnd], yAxe))
                             if yAxe > lastSectionPosition:
                                 lastSectionPosition = yAxe
-                            if element[Element.AddressEnd] > biggestEndAddress:
-                                biggestEndAddress = element[Element.AddressEnd]
-                        elif element[Element.AddressStart] < biggestEndAddress:
-                            for el in plottedElements:
-                                if element[Element.AddressStart] < el[PlottedElement.AddressEnd] and yAxe <= el[PlottedElement.yAxe]:
-                                    yAxe = el[PlottedElement.yAxe] + distanceBetweenElements
+                            if element[Element.addressEnd] > biggestEndAddress:
+                                biggestEndAddress = element[Element.addressEnd]
+                        elif element[Element.addressStart] < biggestEndAddress:
+                            for plottedElement in plottedElements:
+                                if element[Element.addressStart] < plottedElement[PlottedElement.addressEnd] and yAxe <= plottedElement[PlottedElement.yAxe]:
```

Comment:
Alternative option (see issue #59):
```suggestion
                            # Find if a previous element overlaps and increase y level if so
                            for plottedElement in plottedElements:
                                if element[Element.addressStart] < plottedElement[PlottedElement.addressEnd] and yAxe <= plottedElement[PlottedElement.yAxe]:
```
-> you might want to use `any()` here.
-> The alternative solution makes this easier since you only have to look at the last element in the list and do the check (if you do not implement the whole algorithm below).

Alternative solution:

1. Sort (`bisect.insort` package function might be helpful) by end address
2. Iterate over elements (backwards) and check if end address < start address of element to be added
3. If there is a free space insert the new element otherwise add to a new y level

Reply:
The first solution wont work with the current implementation, as we have to know which of plotted elements overlaps the current element